gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/mapper/ImmunizationMapper.java

```diff
@@ -0,0 +1,301 @@
+        } else if (dateText != null) {
+            var noteStringType = new StringType(dateText);
+            return new Annotation(noteStringType);
+        }
+
+        return null;
+    }
+
+    private String getPertinentAnnotation(RCMRMT030101UK04ObservationStatement observationStatement) {
+        if (observationStatement.getPertinentInformation() != null && observationStatement.getPertinentInformation().size() > 0) {
+            var pertinentInformation = observationStatement.getPertinentInformation().get(0);
+
+            if (pertinentInformation.getPertinentAnnotation() != null && pertinentInformation.getPertinentAnnotation().getText() != null) {
+                return pertinentInformation.getPertinentAnnotation().getText();
+            }
+        }
+
+        return null;
+    }
+
+    private Immunization createImmunization(ImmunizationMapperParameters immunizationMapperParameters) {
+        var immunization = new Immunization();
+
+        immunization.getMeta().getProfile().add(new UriType(META_PROFILE));
+        immunization.getIdentifier().add(immunizationMapperParameters.getIdentifierParam());
+        immunization.getExtension().add(immunizationMapperParameters.getVaccineExtensionParam());
+        immunization.getExtension().add(immunizationMapperParameters.getRecordedTimeExtensionParam());
+        immunization
+            .setStatus(Immunization.ImmunizationStatus.COMPLETED)
+            .setNotGiven(false)
+            .setPrimarySource(false)
+            .setPatient(immunizationMapperParameters.getPatientParam())
+            .setId(immunizationMapperParameters.getIdParam());
+
+        if (immunizationMapperParameters.getDateParam() != null) {
+            immunization.setDate(immunizationMapperParameters.getDateParam());
+        }
+        if (immunizationMapperParameters.getEncounterParam() != null) {
+            immunization.setEncounter(immunizationMapperParameters.getEncounterParam());
+        }
+        if (immunizationMapperParameters.getNoteParam() != null) {
+            immunization.getNote().add(immunizationMapperParameters.getNoteParam());
+        }
+        if (immunizationMapperParameters.getPractitionerParam() != null) {
+            immunization.addPractitioner(new Immunization.ImmunizationPractitionerComponent(
+                immunizationMapperParameters.getPractitionerParam()));
+        }
+
+        return immunization;
+    }
+
+    @Getter
+    @Setter
+    public static class ImmunizationMapperParameters {
+        private String idParam;
+        private Identifier identifierParam;
+        private Annotation noteParam;
+        private Extension vaccineExtensionParam;
+        private Date dateParam;
+        private Extension recordedTimeExtensionParam;
+        private Reference patientParam;
+        private Reference encounterParam;
+        private Reference practitionerParam;
+
+        public ImmunizationMapperParameters() {
+
+        }
```

Comment:
Is this needed?

Reply:
> Is this needed?

yea i believe so as this is the constructor that actually allows me to call this class but ill check if there is another way of doing this